internal/buffer/buffer.go

```diff
@@ -126,18 +126,23 @@ type SharedBuffer struct {
 }
 
 func (b *SharedBuffer) insert(pos Loc, value []byte) {
-	b.isModified = true
-	b.HasSuggestions = false
-	b.LineArray.insert(pos, value)
+	if !b.Type.Readonly {
```

Comment:
AFAICS the only reason for this is that `MoveLinesUp` calls this low-level `insert` function directly (which also, like in case with retab, causes another issue: undo doesn't work correctly for MoveLinesUp at the last line).
It was changed to use `insert` instead of `Insert` in the commit  cfc595e. I wonder what was the reason for that.